src/components/multi_select_state.js

```diff
@@ -109,41 +152,76 @@ const withMultiSelectState = WrappedComponent =>
             const index = items.findIndex(item => item.id === id);
             this.setState({ firstItemShiftSelected: index });
           }
+
           const newSelectedItems = this.getNewSelectedItems(id);
-          this.setState({ selectedItems: newSelectedItems }, this.handleChange);
+
+          const newFilteredSelectedItem = newSelectedItems.filter(
+            el => el.id == id
+          );
+
+          this.setState(
+            {
+              selectedItems: newSelectedItems,
+              filteredSelectedItems: [
+                ...filteredSelectedItems,
+                ...newFilteredSelectedItem
+              ]
+            },
+            this.handleChange
+          );
         }
       } else {
         this.unselectItems([id]);
       }
     }
 
     unselectItems(ids) {
-      const { selectedItems } = this.state;
+      const { selectedItems, filteredSelectedItems } = this.state;
+
       const newSelectedItems = selectedItems.filter(
         item => ids.find(id => id === item.id) === undefined
       );
+      const newFilteredSelectedItems = filteredSelectedItems.filter(
+        item => item.id !== ids[0]
```

Comment:
why ids[0]?
unselectItem**s** means that it receives not just one id and the array is here for a reason.